src/shims/os_str.rs

```diff
@@ -0,0 +1,247 @@
+use std::borrow::Cow;
+use std::convert::TryFrom;
+use std::ffi::{OsStr, OsString};
+use std::iter;
+#[cfg(unix)]
+use std::os::unix::ffi::{OsStrExt, OsStringExt};
+#[cfg(windows)]
+use std::os::windows::ffi::{OsStrExt, OsStringExt};
```

Comment:
Please keep these conditional imports visually separated, like they were in the original file.

Reply:
Should it be something like this? (Assuming alphabetical order of imports)
```rust
use std::iter;

#[cfg(unix)]
use std::os::unix::ffi::{OsStrExt, OsStringExt};
#[cfg(windows)]
use std::os::windows::ffi::{OsStrExt, OsStringExt};

use std::path::{Path, PathBuf};
```

Reply:
No, I'd first group all the regular imports (alphabetically sorted, if you want), and then separately the conditional ones.

```rust
use std::iter;
use std::path::{Path, PathBuf};

#[cfg(unix)]
use std::os::unix::ffi::{OsStrExt, OsStringExt};
#[cfg(windows)]
use std::os::windows::ffi::{OsStrExt, OsStringExt};
```